VirtoCommerce.Platform.Data.Azure/AzureBlobProvider.cs

```diff
@@ -73,32 +62,32 @@ public virtual BlobInfo GetBlobInfo(string url)
         /// <summary>
         /// Open blob for read by relative or absolute url
         /// </summary>
-        /// <param name="url"></param>
+        /// <param name="blobUrl"></param>
         /// <returns>blob stream</returns>
-        public virtual Stream OpenRead(string url)
+        public virtual Stream OpenRead(string blobUrl)
         {
-            if (string.IsNullOrEmpty(url))
-                throw new ArgumentNullException(nameof(url));
+            if (string.IsNullOrEmpty(blobUrl))
+                throw new ArgumentNullException(nameof(blobUrl));
 
-            var uri = url.IsAbsoluteUrl() ? new Uri(url) : new Uri(_cloudBlobClient.BaseUri, url.TrimStart('/'));
+            var uri = blobUrl.IsAbsoluteUrl() ? new Uri(blobUrl) : new Uri(_cloudBlobClient.BaseUri, blobUrl.TrimStart('/'));
             var cloudBlob = _cloudBlobClient.GetBlobReferenceFromServer(new Uri(_cloudBlobClient.BaseUri, uri.AbsolutePath.TrimStart('/')));
             return cloudBlob.OpenRead();
         }
 
         /// <summary>
         /// Open blob for write by relative or absolute url
         /// </summary>
-        /// <param name="url"></param>
+        /// <param name="blobUrl"></param>
         /// <returns>blob stream</returns>
-        public virtual Stream OpenWrite(string url)
+        public virtual Stream OpenWrite(string blobUrl)
```

Comment:
It is a general approach.

Reply:
Returned stream is disposable.

Reply:
the following way is better because of you close the unmanaged resource inside in your code and prevent the potential problems from caller code:
public void ReadToStream(string url, Stream destinationStream)
        {
            var cloudBlob = _cloudBlobClient.GetBlobReferenceFromServer(url);
            using (var sourceStream = cloudBlob.OpenRead())
            {
                sourceStream.CopyTo(destinationStream);
            }
        }

Reply:
It is huge refactoring